pkg/config/nodetreemodel/config.go

```diff
@@ -139,18 +154,35 @@ func (c *ntmConfig) Set(key string, newValue interface{}, source model.Source) {
 	switch source {
 	case model.SourceDefault:
 		tree = c.defaults
+	case model.SourceUnknown:
+		tree = c.unknown
 	case model.SourceFile:
 		tree = c.file
+	case model.SourceEnvVar:
+		tree = c.env
+	case model.SourceAgentRuntime:
+		tree = c.runtime
+	case model.SourceLocalConfigProcess:
+		tree = c.localConfigProcess
+	case model.SourceRC:
+		tree = c.remoteConfig
+	case model.SourceFleetPolicies:
+		tree = c.fleetPolicies
+	case model.SourceCLI:
+		tree = c.cli
 	}
 
 	c.Lock()
 
 	previousValue, _ := c.getValue(key)
-	_, _ = c.set(key, newValue, tree, source)
-	updated, _ := c.set(key, newValue, c.root, source)
+
+	parts := strings.Split(strings.ToLower(key), ",")
```

Comment:
You can use `splitKey` here once https://github.com/DataDog/datadog-agent/pull/30835 merges.